vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java

```diff
@@ -668,6 +690,64 @@ public MongoClient dropIndex(String collection, String indexName, Handler<AsyncR
     return this;
   }
 
+  @Override
+  public MongoClient watch(String collection, JsonArray pipeline, WatchOptions options, Handler<AsyncResult<MongoClientChangeStream<MongoClientChange>>> resultHandler) {
+    return createWatchIterable(collection, pipeline, options, false, resultHandler);
+  }
+
+  @Override
+  public MongoClient watchReplaceRoot(String collection, JsonArray pipeline, WatchOptions options, Handler<AsyncResult<MongoClientChangeStream<JsonObject>>> resultHandler) {
+    throw new UnsupportedOperationException();
+    // Not supported until https://jira.mongodb.org/browse/JAVA-2828 is fixed
+    // return createWatchIterable(collection, pipeline, options, true, resultHandler);
+  }
+  
+  @SuppressWarnings("unchecked")
+  private <T> MongoClient createWatchIterable(String collection, JsonArray pipeline, WatchOptions options, boolean replacedRootDocument, Handler<AsyncResult<MongoClientChangeStream<T>>> resultHandler) {
+    requireNonNull(collection);
+    requireNonNull(pipeline);
+
+    if (options == null) {
+      options = new WatchOptions();
+    } else {
+      if (options.getFullDocument() != null
+        && !options.getFullDocument().equals(WatchOptions.FULL_DOCUMENT_UPDATE_LOOKUP)) {
+        resultHandler.handle(Future.failedFuture(new IllegalArgumentException("fullDocument must be null or \"updatedLookup\"")));
```

Comment:
Think the error message makes a bit more sense as:

fullDocument option must be either null or "updatedLookup"